Firestore/core/src/model/server_timestamp_util.cc

```diff
@@ -110,12 +110,12 @@ absl::optional<google_firestore_v1_Value> GetPreviousValue(
       if (IsServerTimestamp(field.value)) {
         return GetPreviousValue(field.value);
       } else {
-        return field.value;
+        return api::optional<google_firestore_v1_Value>(field.value);
       }
     }
   }
 
-  return absl::nullopt;
+  return api::optional<google_firestore_v1_Value>();
```

Comment:
Optional: I believe you could just return `{}` here, for simplicity. Totally up to you.

Reply:
Done.